### PR TITLE
Issue #605: Fix `utils.fmt()` function

### LINKED ISSUE #605: `Utils.fmt()` function does not work properly when there are both underline and color (e.g. `<yellow>..</yellow>`) present
Coloring stops when it stumbles upon underline formatting - '`(.*?)`'

## This:
```
print(fmt(f'<yellow>\nGitHub API returned {err.code} HTTP status with error message: `{err.reason}`. '
          'It looks like the organization or repository name got changed recently and is outdated.\n'
          'Inferring current organization or repository... '
          f'New organization = {current_repo_and_org.split("/")[0]}, '
          f'new repository = {current_repo_and_org.split("/")[1]}.\n'
          'You can update your remote repository via: `git remote set-url <remote_name> <new_repository_url>`. </yellow>'),
      end='',
      flush=True)
```
## produces:

image

while the whole text should be yellow.

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: git_machete/utils.py [hunk 1]
```diff
@@ -237,16 +239,16 @@ class AnsiEscapeCodes:
 
 
 def bold(s: str) -> str:
-    return s if ascii_only or not s else AnsiEscapeCodes.BOLD + s + AnsiEscapeCodes.ENDC
+    return s if ascii_only or not s else AnsiEscapeCodes.BOLD + s + AnsiEscapeCodes.ENDC_BOLD_DIM
```
Comment (PawelLipski) [inline git_machete/utils.py:242]: Nice, I didn't know there's a way to just turn off a single aspect of formatting via ANSI espaces 😯

File: git_machete/utils.py [hunk 2]
```diff
@@ -258,13 +260,14 @@ def colored(s: str, color: str) -> str:
...
 fmt_transformations: List[Callable[[str], str]] = [
+    lambda x: re.sub('`(.*?)`', underline(r"\1"), x),
     lambda x: re.sub('<b>(.*?)</b>', bold(r"\1"), x, flags=re.DOTALL),
     lambda x: re.sub('<u>(.*?)</u>', underline(r"\1"), x, flags=re.DOTALL),
     lambda x: re.sub('<dim>(.*?)</dim>', dim(r"\1"), x, flags=re.DOTALL),
     lambda x: re.sub('<red>(.*?)</red>', colored(r"\1", AnsiEscapeCodes.RED), x, flags=re.DOTALL),
     lambda x: re.sub('<yellow>(.*?)</yellow>', colored(r"\1", AnsiEscapeCodes.YELLOW), x, flags=re.DOTALL),
     lambda x: re.sub('<green>(.*?)</green>', colored(r"\1", AnsiEscapeCodes.GREEN), x, flags=re.DOTALL),
-    lambda x: re.sub('`(.*?)`', r"`\1`" if ascii_only else AnsiEscapeCodes.UNDERLINE + r"\1" + AnsiEscapeCodes.ENDC, x),
+    lambda x: re.sub('<orange>(.*?)</orange>', colored(r"\1", AnsiEscapeCodes.ORANGE), x, flags=re.DOTALL)
```
Comment (PawelLipski) [inline git_machete/utils.py:270]: Has this `<orange>` tag been already used somewhere?

File: tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        Verify behaviour of a Utils fmt() function
+        """
+        os.environ["TERM"] = 'xterm-256color'
+        from git_machete import utils
+        utils.ascii_only = False
+        del os.environ["TERM"]
+
+        input_string = '<red> red <yellow>yellow <b>yellow_bold</b> `yellow_underlined` yellow <green>green </green> default' \
+                       ' <dim> dimmed </dim></yellow> <green>green `green_underlined`</green> default</red>'
+        expected_ansi_string = r'[31m red [33myellow [1myellow_bold[22m [36myellow_underlined[24m yellow [32mgreen ' \
```
Comment (PawelLipski) [inline tests/test_utils.py:16]: :benissimo:

**Review by PawelLipski — APPROVED** [review]

File: git_machete/utils.py (hunk 2 above)
Comment (amalota) [reply git_machete/utils.py:270]: @PawelLipski No it hasn't but if it is available as a color in `AnsiEscapeCodes` then it should be here to, right?

Comment (PawelLipski) [reply git_machete/utils.py:270]: Okay, sure, let's leave

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 - added: `git machete github retarget-pr` now updates annotation for the branch associated with the retargeted PR
 - improved: Modify formatting in error message for `github create-pr`
+- fixed: Modify formatting of `git machete` commands output
```
Comment (PawelLipski) [inline RELEASE_NOTES.md:7]: Nit: lowercase `modify`

**Pull request merged**